index.d.ts

```diff
@@ -576,7 +576,7 @@ declare module 'evergreen-ui' {
   export class Card extends React.PureComponent<CardProps> {
   }
 
-  export interface CheckboxProps extends BoxProps<'input'> {
+  export interface CheckboxProps extends Omit<BoxProps<'input'>, 'innerRef'> {
```

Comment:
I think this should do the trick. It was actually getting caught/highlighted by VS Code directly.